### ISSUE TITLE
Crashes if isort isn't installed

### ISSUE THREAD

**Issue by akaihola** [akaihola/darker#130]
conda-forge/darker-feedstock#1 fails with this error (in the Linux and OSX builds):
```
+ darker --help
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/darker_1619998672207/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/bin/darker", line 7, in <module>
    from darker.__main__ import main
  File "/home/conda/feedstock_root/build_artifacts/darker_1619998672207/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/darker/__main__.py", line 16, in <module>
    from darker.import_sorting import apply_isort, isort
  File "/home/conda/feedstock_root/build_artifacts/darker_1619998672207/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/darker/import_sorting.py", line 22, in <module>
    isort_code = getattr(isort, "code")
AttributeError: 'NoneType' object has no attribute 'code'
```

**Comment by akaihola** [comment]
@martinRenou, I'm taking a look at this now.

**Comment by akaihola** [comment]
This is reproducible in any environment:
```python
$ pip uninstall -y isort
$ darker --help
Traceback (most recent call last):
  File "bin/darker", line 33, in <module>
    sys.exit(load_entry_point('darker', 'console_scripts', 'darker')())
  File "bin/darker", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "darker/__main__.py", line 16, in <module>
    from darker.import_sorting import apply_isort, isort
  File "darker/import_sorting.py", line 22, in <module>
    isort_code = getattr(isort, "code")
AttributeError: 'NoneType' object has no attribute 'code'
```

**Comment by martinRenou** [comment]
I am fixing the conda package by adding a dependency on isort: https://github.com/conda-forge/darker-feedstock/pull/2

What do you think? Do you think we should do the same in your `setup.py` file so that pip picks it up as well?

**Comment by akaihola** [comment]
@martinRenou, `isort` is an optional dependency which is required only when `darker --isort` is used. You can install darker with isort by doing `pip install 'darker[isort]'` (or, from a local source tree, `pip install '.[isort]'`.

**Comment by martinRenou** [comment]
I see.

So having the isort dependency would temporarily fix the conda package for 1.2.3, but I guess we should remove it once the bug is fixed?

**Comment by akaihola** [comment]
@martinRenou, when running the test suite, `isort` should always be installed.

There are specific tests which mock `sys.modules` and remove `isort` temporarily to check that all functionality (besides `--isort`) still would works if `isort` wasn't installed.
